pychromecast/dial.py

```diff
@@ -57,6 +57,10 @@ def _get_status(host, services, zconf, path):
                 _LOGGER.debug("Resolved service %s to %s", service, host)
                 break
 
+    # Reinitialize the session in case the device was rebooted
+    CC_SESSION = requests.Session()
```

Comment:
All this does is create a local variable (hence the CI errors, I believe). You need `global CC_SESSION` above this line to reassign the global.

Reply:
Thanks @skorokithakis, I updated the patch

Reply:
That looks much better, the CI failure now looks like a linter problem, to me. At least, I don't see how this can be done without using the `global` statement.

Reply:
I think that we can get rid of the CC_SESSION completely. There is no reason to keep one around ?

Reply:
@balloob Hi, CC_SESSION is used for .get() in _get_status() and .post() in reboot(). What would be the alternative if there is no CC_SESSION for .get() and.post()?

Reply:
@balloob [This also works](http://ix.io/1RBj) which changes CC_SESSION to a local variable that is initialized with the session each time it is used.

Reply:
Well I was thinking why wouldn't we just use `requests.get` directly ? What is in the new session that fixes it?

Reply:
If I try to print CC_CESSION before and after it's set to `requests.Session()` and it says `UnboundLocalError: local variable 'CC_SESSION' referenced before assignment` for the before statement, in _get_status. If I try `requests.get` it silently fails. .get() is a method on `request.Session()`, not `request`. `request.Session().get` also fails. It needs a variable to assign `request.Session()`, then needs `.headers['content-type'] = 'application/json'` called on it, then it works. This requires a variable unless you blindly call `request.Session()` twice, once to set json and once for `get/post`, but this also fails anyway. As far as I can tell, a variable is needed regardless.

Reply:
requests.get can have a headers dictionary be passed in. https://2.python-requests.org/en/master/user/quickstart/#custom-headers

Reply:
@balloob Yes that does work, thanks. I have updated the patch.